### PR TITLE
fix listener access log test

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/39619

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/accesslog_test.go [hunk 1]
```diff
@@ -104,7 +103,7 @@ func TestListenerAccessLog(t *testing.T) {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			// Update MeshConfig
-			m := mesh.DefaultMeshConfig()
+			m := &meshconfig.MeshConfig{}
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/accesslog_test.go:106]: I am not sure I understand this. Can you explain why this fixes it?

It seems like the issue is that we have a shared cache, so maybe we just need `accessLogBuilder.reset()` before the test (or maybe in `buildListeners`, etc)?

**Comment by ramaraochavali** [comment]
@howardjohn @hzxuzhonghu PTAL

File: pilot/pkg/networking/core/v1alpha3/accesslog_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/accesslog_test.go:106]: IIUC, this is not to mutate the default config. I do not look into the original issue, but this way it makes no harm.

@howardjohn to check whether it fixes original issue?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/accesslog_test.go:106]: yes. The issue is we are mutating default config in the test which other tests are dependent on.

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/accesslog_test.go:106]: I have run multiple times with shuffle on - it did not fail.

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/accesslog_test.go:106]: How are we mutating it? DefaultMeshConfig is a new copy.

I know this fixes it but want tounderstand why

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/accesslog_test.go:106]: > DefaultMeshConfig is a new copy.

Ah .. Let me debug a bit a more on how this is working :-)

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/accesslog_test.go]: Ok. this is actually not working. When we pass `&meshconfig.MeshConfig{}`, no listeners are constructed as `proxyListenPort` is zero. So it appears to pass but is incorrect. I fixed it now. Basically the accesslog cache  should be reset before buildListener call not after.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**